pkg/api/datasources.go

```diff
@@ -280,6 +280,11 @@ func (hs *HTTPServer) UpdateDataSource(c *models.ReqContext) response.Response {
 	datasourcesLogger.Debug("Received command to update data source", "url", cmd.Url)
 	cmd.OrgId = c.OrgId
 	cmd.Id = c.ParamsInt64(":id")
+
+	if cmd.Url == "" {
```

Comment:
Should we consider adding a quick unit test which exercises this case? Especially since frontend behavior now depends on it.

Reply:
Will this be run for all datasources? I don't think all datasources specifies this value hence can't be required.

Would it be possible to enable a datasource to specify what values they are expecting and then validate according to those? @marefr is this something we have in place today?

Reply:
Ah, so I think the previous code looks correct. Basically it uses the `datasource` to validate the URL.

So I think you should check how this is done for Prometheus and Loki to fix the underlying issue.

Reply:
Thanks for the info! This PR is on hold for now since I'm working on higher priority issues, but I was starting to have failing tests because of this.

Reply:
OK ok, let us know if you want to jump on a call and try to figure out an alternative approach to this solution.